### ISSUE TITLE
python3 issue with spellchecker

### ISSUE THREAD

**Issue by gabn88** [aljosa/django-tinymce#153]
I got this issue: 
TypeError("the JSON object must be str, not 'bytes'",)

It is in the view spell_check.

doing request.body.decode('utf-8')  generates ValueError: Expecting value: line 1 column 1 (char 0)
on the json.loads. 

My possible fix:
def spell_check(request):
    """
    Returns a HttpResponse that implements the TinyMCE spellchecker protocol.
    """
    try:
        import enchant

        arg = request.POST['text']
        method = request.POST['method']
        lang = request.POST['lang']

        if not enchant.dict_exists(str(lang)):
            raise RuntimeError("dictionary not found for language '%s'" % lang)

        checker = enchant.Dict(str(lang))

        if method == 'spellcheck':
            result = [word for word in arg if word and not checker.check(word)]
        elif method == 'getSuggestions':
            result = checker.suggest(arg)
        else:
            raise RuntimeError("Unkown spellcheck method: '%s'" % method)
        output = {
            'result': result,
            'error': None,
        }
    except Exception as e:
        logging.exception("Error running spellchecker")
        return HttpResponse(_("Error running spellchecker"))
    return HttpResponse(json.dumps(output),
            content_type='application/json')

But it does not work.

**Comment by romanvm** [comment]
I don't know about v.3 but TinyMCE 4 may send either POST data or JSON-RPC http payload depending on JavaScript callback implementation.

Currently I'm playing with django-tinymce4 and managed to fix almost everything, except gzip.
This is my working implementation of spellcheck (uses JSON-RPC) for TinyMCE 4:
```Python
def spell_check(request):
    """
    Returns a HttpResponse that implements the TinyMCE spellchecker protocol.
    """
    data = json.loads(request.body.decode('utf-8'))
    output = {'id': data['id']}
    error = None
    try:
        import enchant
        from enchant.checker import SpellChecker
        if data['params']['lang'] not in enchant.list_languages():
            raise RuntimeError
        checker = SpellChecker(data['params']['lang'])
        checker.set_text(strip_tags(data['params']['text']))
        output['result'] = {checker.word: checker.suggest() for err in checker}
    except ImportError:
        error = 'pyenchant package is not installed!'
        logging.exception(error)
    except RuntimeError:
        error = 'Missing dictionary {0}!'.format(data['params']['lang'])
        logging.exception(error)
    except Exception:
        error = 'Unknown error!'
        logging.exception(error)
    if error is not None:
        output['error'] = error
    return HttpResponse(json.dumps(output), content_type='application/json')
```
This is a template for JavaScript spellcheck callback, if you are interested:
```JavaScript
function(method, text, success, failure) {
  tinymce.util.JSONRequest.sendRPC(
  {
    url: '{% url "tinymce-spellchecker" %}',
    method: 'spellcheck',
    params: {
      lang: this.getLanguage(),
      text: text
    }, // end params
    success: function(result) {
      success(result);
    }, // end success
    error: function(error, xhr) {
      failure('Spellcheck error: ' + error);
    } // end error
  }); // End sendRPC
} // end spellchecker_callback
```

**Comment by vanadium23** [comment]
Fix for this is in master. We'll release new version soonish.

**Comment by vanadium23** [comment]
We've released 2.3.0 - try to check this bug.
